exporter/otlpexporter/cfgMeta.yaml

```diff
@@ -0,0 +1,162 @@
+  type: exporterhelper.RetrySettings
+  kind: struct
+  fields:
+  - name: enabled
+    kind: bool
+    default: true
+    doc: |
+      Enabled indicates whether to not retry sending batches in case of export failure.
+  - name: initial_interval
+    type: time.Duration
+    kind: int64
+    default: 5s
+    doc: |
+      InitialInterval the time to wait after the first failure before retrying.
+  - name: max_interval
+    type: time.Duration
+    kind: int64
+    default: 30s
+    doc: |
+      MaxInterval is the upper bound on backoff interval. Once this value is reached the delay between
+      consecutive retries will always be `MaxInterval`.
+  - name: max_elapsed_time
+    type: time.Duration
+    kind: int64
+    default: 5m0s
+    doc: |
+      MaxElapsedTime is the maximum amount of time (including retries) spent trying to send a request/batch.
+      Once this value is reached, the data is discarded.
+- name: endpoint
+  kind: string
+  doc: |
+    The target to which the exporter is going to send traces or metrics,
+    using the gRPC protocol. The valid syntax is described at
+    https://github.com/grpc/grpc/blob/master/doc/naming.md.
+- name: compression
+  kind: string
+  doc: |
+    The compression key for supported compression types within
+    collector. Currently the only supported mode is `gzip`.
+- name: ca_file
+  kind: string
+  doc: |
+    Path to the CA cert. For a client this verifies the server certificate.
+    For a server this verifies client certificates. If empty uses system root CA.
+    (optional)
+- name: cert_file
+  kind: string
+  doc: |
+    Path to the TLS cert to use for TLS required connections. (optional)
+- name: key_file
+  kind: string
+  doc: |
+    Path to the TLS key to use for TLS required connections. (optional)
+- name: insecure
+  kind: bool
+  default: false
+  doc: |
+    In gRPC when set to true, this is used to disable the client transport security.
+    See https://godoc.org/google.golang.org/grpc#WithInsecure.
+    In HTTP, this disables verifying the server's certificate chain and host name
+    (InsecureSkipVerify in the tls Config). Please refer to
+    https://godoc.org/crypto/tls#Config for more information.
+    (optional, default false)
+    TODO(ccaraman): With further research InsecureSkipVerify is a valid option
+    for gRPC connections. Add that ability to the TLSClientSettings in a subsequent
+    pr.
+- name: server_name_override
+  kind: string
+  doc: |
+    ServerName requested by client for virtual hosting.
+    This sets the ServerName in the TLSConfig. Please refer to
+    https://godoc.org/crypto/tls#Config for more information. (optional)
+- name: keepalive
+  type: '*configgrpc.KeepaliveClientConfig'
+  kind: ptr
+  doc: |
+    The keepalive parameters for gRPC client. See grpc.WithKeepaliveParams
+    (https://godoc.org/google.golang.org/grpc#WithKeepaliveParams).
+  fields:
+  - name: time
+    type: time.Duration
+    kind: int64
+    meta: omitempty
```

Comment:
What's the meta field supposed to contain? Don't understand why it has omitempty.

Reply:
Is it supposed to be tags? Seems like maybe it should be something like this if so:

```yaml
meta:
  yaml: ",omitempty"
  mapstructure: "server_name_override,omitempty"
```

(Just to make up two tags as examples)

Reply:
I think we might want the tags but could also add in future if needed. For example if we add validators using tags like:

```
endpoint string `validate:nonempty`
```

You might want it for docs.

I was also going to say we would want it for doing default values like we do in smart agent but in otel it uses the factory instance to populate default values. Can be a followup as this PR is getting large but ideally we would populate the fields here with those default values if one is set.

Reply:
The structtag library calls these 'options'. I left them out for now because I'm not sure how they'll be useful with the current values (apparently, we only use 'omitempty'). I figure we can add it back in if/when a use arises. If you still would prefer these in now though, I can add them in the implementation PR.